### PR TITLE
Add Windows support for the ssh plugin

### PR DESCRIPTION
**Pull request by Miauwkeru**
- Move os/unix/ssh.py to apps/ssh/openssh.py
- Create tests for windows version of openssh
- Replace C:\\ with sysvol in output for some tests
- Abstract looking for sshd directory
- rename record descriptor names

(DIS-1854)

### CODE REVIEW

File: dissect/target/plugins/apps/ssh/openssh.py
Comment (Schamper) [inline dissect/target/plugins/apps/ssh/openssh.py:1]: Can you add a `__init__.py` to the `ssh` directory?

**Review by Schamper — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/apps/ssh/openssh.py [hunk 1]
```diff
@@ -64,7 +65,13 @@ class SSHPlugin(Plugin):
     __namespace__ = "ssh"
 
     def check_compatible(self):
-        return len(list(self.target.users())) > 0 and self.target.os != "windows"
+        return len(list(self.target.users())) > 0 or self._sshd_directory().exists()
+
+    def _sshd_directory(self) -> TargetPath:
```
Comment (pyrco) [inline dissect/target/plugins/apps/ssh/openssh.py:70]: Would it be useful to make this an `@property`? And then also possibly caching it by saving it to some instance variable.

**Review by cecinestpasunepipe — CHANGES_REQUESTED** [review]

File: dissect/target/plugins/apps/ssh/openssh.py (hunk 1 above)
Comment (pyrco) [reply dissect/target/plugins/apps/ssh/openssh.py]: Even better  :laughing:

File: dissect/target/plugins/apps/ssh/openssh.py [hunk 2]
```diff
@@ -64,7 +66,14 @@ class SSHPlugin(Plugin):
     __namespace__ = "ssh"
 
     def check_compatible(self):
-        return len(list(self.target.users())) > 0 and self.target.os != "windows"
+        return len(list(self.target.users())) > 0 or self.sshd_directory.exists()
```
[the remark this answers is not in the document]
Comment (pyrco) [reply dissect/target/plugins/apps/ssh/openssh.py:69]: You want to check for both client and server dirs, but technically the users check doesn't really do the former..

File: dissect/target/plugins/apps/ssh/openssh.py [hunk 3]
```diff
@@ -64,7 +66,14 @@ class SSHPlugin(Plugin):
...
+
+    @cached_property
+    def sshd_directory(self) -> TargetPath:
+        if (target_path := self.target.fs.path("/sysvol/ProgramData/ssh")).exists():
```
[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/target/plugins/apps/ssh/openssh.py]: Ok, nice.

File: dissect/target/plugins/apps/ssh/openssh.py [hunk 4]
```diff
@@ -64,7 +66,14 @@ class SSHPlugin(Plugin):
...
+            return target_path
+
+        return self.target.fs.path("/etc/ssh/")
 
     @export(record=AuthorizedKeysRecord)
     def authorized_keys(self) -> Iterator[AuthorizedKeysRecord]:
```
[the remark this answers is not in the document]
Comment (cecinestpasunepipe) [reply dissect/target/plugins/apps/ssh/openssh.py:94]: Can you still update the docstrings? The output now still says it is UNIX-only.

**Review by cecinestpasunepipe — CHANGES_REQUESTED** [review]

**Review by cecinestpasunepipe — APPROVED** [review]

**Comment by cecinestpasunepipe** [comment]
LGTM,

Still please don't forget to address @pyrco 's change requests. But regarding my suggestions, I am OK with it.

**Review by pyrco — APPROVED** [review]

**Pull request merged**